lib/charms/opensearch/v0/helper_conf_setter.py

```diff
@@ -42,10 +44,11 @@ def __str__(self):
         return self.value
 
 
-class YamlConfigSetter:
-    """Utility class for updating YAML config, supporting diverse object types and nestedness.
+class ConfigSetter(ABC):
+    """Base class for manipulating YAML Config, of multiple types and any depth level.
+
+    conf_setter = YamlConfigSetter() or another config setter
 
-    conf_setter = YamlConfigSetter()
     put("file.yml", "a.b", "new_name")
     put("file.yml", "a.b/c.obj/key3/key1.a/obj", {"a": "new_name_1", "b": ["hello", "world"]})
```

Comment:
The `/` is the delimiter / separator character by default. You can refer to the `sep="/"` argument of the `ConfigSetter.put(...)` method.
The reason I use the `.` characters in my tests, is because opensearch uses many key names with the said character. i.e: `cluster.initial_cluster_manager_nodes` which should be treated as a simple key name.

This is a utility class, that should handle any delimiter character for nested content (`sep` argument).

Reply:
Yes, that's what I thought. However to make this clear (without requiring the user to think and understand), one could either show the examples you provided in the answer or even not use complicated strings that may create confusion. In the tests, you generally don't use keys with ".", which is way more clear, from a test and user standpoint.